### ISSUE TITLE
Xcode 15.3 RC2 App installation failed

### ISSUE THREAD

**Issue by muhasturk** [googleads/swift-package-manager-google-user-messaging-platform#8]
Xcode 15.3 RC2 failed to load app to Simulator points to `UserMessagingPlatform.framework/Info.plist: No such file or directory`

Yes, I already cleaned derived data. Xcode 15.2 builds and installs without any problem.

```log
App installation failed: Unable to Install “”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework; Extra info about "/Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist": Couldn't stat /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist: No such file or directory
User Info: {
    DVTErrorCreationDateKey = "2024-03-04 19:51:24 +0000";
    IDERunOperationFailingWorker = IDELaunchiPhoneSimulatorLauncher;
    SimCallingSelector = "installApplication:withOptions:error:";
}
--
App installation failed: Unable to Install “”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework; Extra info about "/Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist": Couldn't stat /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist: No such file or directory
User Info: {
    IDERunOperationFailingWorker = IDELaunchiPhoneSimulatorLauncher;
    SimCallingSelector = "installApplication:withOptions:error:";
}
--
Unable to Install “”
Domain: IXUserPresentableErrorDomain
Code: 1
Failure Reason: Please try again later.
Recovery Suggestion: Failed to load Info.plist from bundle at path /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework; Extra info about "/Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist": Couldn't stat /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist: No such file or directory
--
Failed to load Info.plist from bundle at path /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework; Extra info about "/Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist": Couldn't stat /Users/h/Library/Developer/CoreSimulator/Devices/027750AB-651D-494D-9232-15046190BEBC/data/Library/Caches/com.apple.mobile.installd.staging/temp.GAIDUw/extracted/X.app/Frameworks/UserMessagingPlatform.framework/Info.plist: No such file or directory
Domain: MIInstallerErrorDomain
Code: 35
User Info: {
    FunctionName = "-[MIBundle _validateWithError:]";
    LegacyErrorString = PackageInspectionFailed;
    SourceFileLine = 64;
}
--

Event Metadata: com.apple.dt.IDERunOperationWorkerFinished : {
    "device_model" = "iPhone15,4";
    "device_osBuild" = "17.4 (21E213)";
    "device_platform" = "com.apple.platform.iphonesimulator";
    "dvt_coredevice_version" = "355.24";
    "dvt_mobiledevice_version" = "1643.100.58";
    "launchSession_schemeCommand" = Run;
    "launchSession_state" = 1;
    "launchSession_targetArch" = arm64;
    "operation_duration_ms" = 154;
    "operation_errorCode" = 1;
    "operation_errorDomain" = IXUserPresentableErrorDomain;
    "operation_errorWorker" = IDELaunchiPhoneSimulatorLauncher;
    "operation_name" = IDERunOperationWorkerGroup;
    "param_debugger_attachToExtensions" = 0;
    "param_debugger_attachToXPC" = 1;
    "param_debugger_type" = 3;
    "param_destination_isProxy" = 0;
    "param_destination_platform" = "com.apple.platform.iphonesimulator";
    "param_diag_MainThreadChecker_stopOnIssue" = 0;
    "param_diag_MallocStackLogging_enableDuringAttach" = 0;
    "param_diag_MallocStackLogging_enableForXPC" = 1;
    "param_diag_allowLocationSimulation" = 1;
    "param_diag_checker_tpc_enable" = 1;
    "param_diag_gpu_frameCapture_enable" = 0;
    "param_diag_gpu_shaderValidation_enable" = 0;
    "param_diag_gpu_validation_enable" = 0;
    "param_diag_memoryGraphOnResourceException" = 0;
    "param_diag_queueDebugging_enable" = 1;
    "param_diag_runtimeProfile_generate" = 0;
    "param_diag_sanitizer_asan_enable" = 0;
    "param_diag_sanitizer_tsan_enable" = 0;
    "param_diag_sanitizer_tsan_stopOnIssue" = 0;
    "param_diag_sanitizer_ubsan_stopOnIssue" = 0;
    "param_diag_showNonLocalizedStrings" = 0;
    "param_diag_viewDebugging_enabled" = 1;
    "param_diag_viewDebugging_insertDylibOnLaunch" = 1;
    "param_install_style" = 0;
    "param_launcher_UID" = 2;
    "param_launcher_allowDeviceSensorReplayData" = 0;
    "param_launcher_kind" = 0;
    "param_launcher_style" = 0;
    "param_launcher_substyle" = 0;
    "param_runnable_appExtensionHostRunMode" = 0;
    "param_runnable_productType" = "com.apple.product-type.application";
    "param_structuredConsoleMode" = 1;
    "param_testing_launchedForTesting" = 0;
    "param_testing_suppressSimulatorApp" = 0;
    "param_testing_usingCLI" = 0;
    "sdk_canonicalName" = "iphonesimulator17.4";
    "sdk_osVersion" = "17.4";
    "sdk_variant" = iphonesimulator;
}
```

### System Information

- macOS Version 14.2.1 (Build 23C71)
- Xcode 15.3 (22618) (Build 15E204a)
- Timestamp: 2024-03-04T22:51:24+03:00

**Comment by kenzo-ai** [comment]
Same issue. Any suggestions or workarounds?

**Comment by zeusent** [comment]
Same issue here. This package is a dependency for https://github.com/googleads/swift-package-manager-google-mobile-ads which has the same issue (https://github.com/googleads/swift-package-manager-google-mobile-ads/issues/62). Is this going to be fixed soon?

**Comment by malandr2** [comment]
See https://github.com/googleads/swift-package-manager-google-mobile-ads/issues/62 for the ongoing discussion and upcoming resolution.

**Comment by malandr2** [comment]
We expect to have a new UMP SDK version released next week that supports Xcode 15.3 using the workaround

https://github.com/googleads/swift-package-manager-google-mobile-ads/issues/62 for context

**Comment by malandr2** [comment]
@all

We have just released SPM version 2.3.0 which should address the 15.3 Xcode issue. Please report back the issue still persists. Thanks!

**Comment by malandr2** [comment]
Closing this issue as it's been reported fixed.
